File: opus-mk-ios/Opus MK/Opus MK/Lib/VenueLabel.swift
import Foundation

enum VenueLabel {
    static func text(for listing: PublishedListing) -> String {
        if listing.industry == "hospitality" {
            let type = listing.venueType?
                .replacingOccurrences(of: "_", with: " ")
                .capitalized ?? "Restaurant"
            if let cuisine = listing.cuisine?.first, !cuisine.isEmpty {
                return "\(type) · \(cuisine)"
            }
            return type
        }
        return BeautyCategories.label(for: listing.beautyCategory)
            ?? listing.beautyCategory?
                .replacingOccurrences(of: "_", with: " ")
                .capitalized
            ?? "Beauty"
    }
}

File: opus-native/src/lib/discover/venue-label.ts
import { beautyCategoryLabel } from '@/lib/discover/beauty-categories';
import type { PublishedListing } from '@/lib/discover/types';

export function venueLabel(listing: PublishedListing): string {
  if (listing.industry === 'hospitality') {
    const type =
      listing.venueType?.replace(/_/g, ' ').replace(/\b\w/g, (c) => c.toUpperCase()) ??
      'Restaurant';
    const cuisine = listing.cuisine?.[0];
    if (cuisine) return `${type} · ${cuisine}`;
    return type;
  }

  return (
    beautyCategoryLabel(listing.beautyCategory) ??
    listing.beautyCategory?.replace(/_/g, ' ').replace(/\b\w/g, (c) => c.toUpperCase()) ??
    'Beauty'
  );
}
